.github/workflows/ci.yml

```diff
@@ -0,0 +1,134 @@
+## (5b) Pin to commits, not tags
+
+name: CI Build and Check
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+env:
+  IMAGE: rancher/kube-api-auth
+
+permissions:
+  contents: read
+  security-events: write # upload Trivy Sarif results
+
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Get Sources
+      uses: actions/checkout@v3
+
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: '1.22'
+        cache: false
+
+    - name: Tidy and verify modules
+      run: |
+        echo Tidying up modules
+        go mod tidy
+        echo Verifying modules
+        go mod verify
```

Comment:
I'm a bit worried about tidying the modules. I think this will "fix" issues, if any. And we would not be able to see it.

Probably we should add [this](https://github.com/golang/go/issues/27005), when ready.

Reply:
See above, regarding copying `scripts/validate`.

Reply:
Comment added with commit 1c97df6fcea44baeb3d1fde45b761c49f65e5628

Reply:
Hm, I was having a look again at this, and there is a piece missing that was doing exactly the check that I was talking about.

If you look at the `validate` script there is a check for the status of untracked files after the `go mod tidy`, to validate the correctness of the committed modules: https://github.com/rancher/kube-api-auth/blob/b3dd13ce2f9a574ddfff154839b06e8574e440b4/scripts/validate#L14-L17

Reply:
I saw this part and removed it because I absolutely did not expect to see untracked files in a CI checkout from the repository.

That said, it should not hurt to be run, and if we do then calling `scripts/validate` is certainly simpler.
Note that using `make validate` will invoke Dapper.

Reply:
Good, `validate` works. As expected the `golangci-lint` direct call in the script is not done, the binary is not in the PATH.